_pages/documentation/gem5-stdlib/7-using-local-data-sources.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: documentation
+title: Using local Data sources in gem5
+parent: gem5-standard-library
+doc: gem5 documentation
+permalink: /documentation/gem5-stdlib/using-local-resources
+author: Harshil Patel
+---
+
+gem5 supports using local data sources in the form of a MongoDB Atlas and JSON datasource. To utilize data sources other than the main gem5 resources database, you will need to override the gem5-resources-config.
+
+MongoDB Atlas Config Format:
+```json
+{
+    "sources":{
+        "example-atlas-config": {
+            "dataSource": "datasource name",
+            "database": "database name",
+            "collection": "collection name",
+            "url": "Atlas data API URL",
+            "authUrl": "Atlas authentication URL",
+            "apiKey": "API key for data API for MongoDB Atlas",
+            "isMongo": true
+        }       
+    }
+}
+```
+JSON Config Format:
+
+```json
+{
+    "sources":{
+        "example-json-config": {
+            "url": "local path to JSON file or URL to a JSON file",
+            "isMongo": false
+        }
+    }
+}
+```
+### Updating gem5 Resources Config
+
+You can update the gem5 resources config in the following ways:
+
+ - Set the GEM5_CONFIG environment variable to point to a new config file. This will override the default config.
+
+- Have a file named gem5-config.json in the current working directory. This will also override the default config.
+
+- If the above two methods are not used, then the default resources config will be used.
+
+You can also override the resources config with a JSON file or add a JSON file to the config as follows:
+
+- Use the GEM5_RESOURCE_JSON environment variable to override the default config to use a JSON file.
+
+- GEM5_RESOURCE_JSON_APPEND environment variable to add a JSON file to the current config. 
+
+If the gem5 resources config was updated by the GEM5_CONFIG environment variable or having a gem5-config.json in current working directory, then these flags will override or add to those configs.
+
```

Comment:
I think this is confusing. At least it was to me, and i'm supposed to know this. As a side note: I'm starting to regret supporting both these cases because it's getting messy with all this "if this then that but not in the case of x where then default z is overridden".

To aid understanability I'd create two completely seperate sections "Add a MongoDB data sources" and "Add a JSON file data source".

Reply:
The above three bullet points (points before line 50)  works with both JSON and MongoDB database or a combination of them but the later two are specifically for one JSON. I can clarify the use case for the later points is that you have a local JSON which you want to use to run gem5.

Reply:
Ah ok. if you can, just try to separate these things off into some logical sections. I realize there's overlap here and I trust you to handle it. I just don't want to confuse users by talking about two things at once or jumping to face between explaining the two separate data sources.